rectools/models/base.py

```diff
@@ -398,7 +398,8 @@ def recommend(
         """
         self._check_is_fitted()
         self._check_k(k)
-
+        # `dataset.item_id_map.external_dtype` can change
+        original_item_type = dataset.item_id_map.external_dtype
```

Comment:
Let's do the same logic for users. Let's assume that users dtype could also be broken during `_custom_transform_dataset_u2i`

Reply:
fixed